public/app/features/alerting/unified/home/Insights.tsx

```diff
@@ -98,12 +98,22 @@ const namespace = config.bootData.settings.namespace;
 
 export const INSTANCE_ID = namespace.includes('stack-') ? namespace.replace('stack-', '') : undefined;
 
-export function getInsightsScenes() {
+export const getInsightsDataSources = () => {
   const dataSourceSrv = getDataSourceSrv();
 
   [ashDs, cloudUsageDs, grafanaCloudPromDs].forEach((ds) => {
     ds.settings = dataSourceSrv.getInstanceSettings(ds.uid);
   });
+  return [ashDs, cloudUsageDs, grafanaCloudPromDs];
+};
+
+export const insigthsIsAvailable = () => {
+  const [_, cloudUsageDs, __] = getInsightsDataSources();
```

Comment:
Is there a particular reason why we check for the cloud usage one specifically instead of something like `getInsightsDataSources.some(dataSource => Boolean(dataSource.settings))`?

Reply:
I noticed that the[ only required data source](https://github.com/grafana/grafana/blob/13a4a46cc10ef9e2e6bd06c2c91215d3101ce98c/public/app/features/alerting/unified/home/Insights.tsx#L110-L114) is `cloudUsageDs` 👀

Reply:
But then inside `getMimirManagedRulesScenes`, it uses `grafanaCloudPromDs` 🤔 So I think the code is making assumptions about datasources being present based on others perhaps?

Reply:
yes, but it only shows this scene if it also has `cloudUsageDs.settings`. So `cloudUsageDs.settings` is the minimum requirement for showing an Insights panel

Reply:
I updated the description in the PR